### PR TITLE
IGNITE-16558 Sql schema synchronization

### PR DESCRIPTION
**Pull request by korlov42**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java [hunk 1]
```diff
@@ -72,36 +79,73 @@ public SchemaPlus schema(@Nullable String schema) {
     /** {@inheritDoc} */
     @Override
     @NotNull
```
Comment (tledkov-gridgain) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java:74]: - Two spaces between 'requiredVer={}' and 'latestKnownVer'
- Should we use different exceptions for cases when table not found (normal flow, e.g. table dropped) and invalid schema version (breaks internal invariant)

Comment (tledkov-gridgain) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java:74]: - Two spaces between 'requiredVer={}' and 'latestKnownVer'
- Should we use different exceptions for cases when table not found (normal flow, e.g. table dropped) and invalid schema version (breaks internal invariant)

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java [hunk 2]
```diff
@@ -72,33 +72,43 @@ public SchemaPlus schema(@Nullable String schema) {
...
         }
 
         if (table == null) {
             throw new IgniteInternalException(
                     IgniteStringFormatter.format("Table not found [tableId={}]", id));
         }
 
+        if (table.version() < ver) {
+            throw new IgniteInternalException(
+                    IgniteStringFormatter.format("Table version not found [tableId={}, requiredVer={},  latestKnownVer={}]",
```
Comment (tledkov-gridgain) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java:92]: - Two spaces between 'requiredVer={}' and 'latestKnownVer'
- Should we use different exceptions for cases when table not found (normal flow, e.g. table dropped) and invalid schema version (breaks internal invariant)

Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java:92]: > Should we use different exceptions

I agree that "Table version not found" is more like `IllegalStateException`, but currently they both will be handled in the same way. With that said, do we really need a separate exception right now?

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/IgniteTableModify.java [hunk 3]
```diff
@@ -117,6 +117,8 @@ public IgniteRel clone(RelOptCluster cluster, List<IgniteRel> inputs) {
     public RelWriter explainTerms(RelWriter pw) {
         return super.explainTerms(pw)
                 .itemIf("tableId", getTable().unwrap(InternalIgniteTable.class).id().toString(),
+                        pw.getDetailLevel() == ALL_ATTRIBUTES)
```
Comment (tledkov-gridgain) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/IgniteTableModify.java:120]: Why `ProjectableFilterableTableScan#explainTerms` uses condition `pw.getDetailLevel() != EXPPLAN_ATTRIBUTES` but here `pw.getDetailLevel() == ALL_ATTRIBUTES`?

**Review by tledkov-gridgain — APPROVED** [review]
OK with me.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/IgniteTableModify.java (hunk 3 above)
Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/IgniteTableModify.java:120]: It's kinda mess here. In fact, we are using `ALL_ATTRIBUTES` in plan serialisation as well as in plan explanation. So, I'he changed `ProjectableFilterableTableScan#explainTerms` to be consistent with TableModify rel.

**Pull request merged**